test/functional/testlib.bash

```diff
@@ -2313,13 +2322,47 @@ generate_test() { # swupd_function
 # tests since all tests require at least the creation of a  test environment
 #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
+global_setup_used=true
+global_env=false
+
 setup() {
 
-	# the first time setup runs, run the global_setup
+	# a global environment will always use number 1, so check for that
+	# environment first
+	if [ -d "$TEST_NAME"_1 ]; then
+		# if a global environment is found we want to continue using that
+		# as our test environment for the rest of the tests, if not continue
+		# numbering the envs according to the test number
+		if [ -e "$TEST_NAME"_1/.global_env ]; then
```

Comment:
If I understand this correctly, only 1 test environment will be saved when using a global test setup, even when multiple tests exist. In this case, would it make sense to copy the state of "$TEST_NAME" to "$TEST_NAME"_"$BATS_TEST_NUMBER" before the teardown step for each test? This would save the state of each test before tearing it down and starting the next one

Reply:
I guess we can do that @jwakre, I'm not sure how useful can that be but it should not hurt either. I'll include that.